### ISSUE TITLE
Trivial patch needed for GCC 13

### ISSUE THREAD

**Issue by hobbes1069** [AcademySoftwareFoundation/openexr#1327]
As gcc gets more pedantic over time apparently you don't get some `int` data types for free anymore:

```
/builddir/build/BUILD/openexr-3.1.5/src/bin/exrcheck/main.cpp: In function 'bool exrCheck(const char*, bool, bool, bool, bool)':
/builddir/build/BUILD/openexr-3.1.5/src/bin/exrcheck/main.cpp:59:13: error: 'uintptr_t' does not name a type
   59 |       const uintptr_t kMaxSize = uintptr_t(-1) / 4;
      |             ^~~~~~~~~
/builddir/build/BUILD/openexr-3.1.5/src/bin/exrcheck/main.cpp:16:1: note: 'uintptr_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   15 | #include <vector>
  +++ |+#include <cstdint>
   16 | 
```

And the trivial patch not worth a PR:
```
Index: openexr-3.1.5/src/bin/exrcheck/main.cpp
===================================================================
--- openexr-3.1.5.orig/src/bin/exrcheck/main.cpp
+++ openexr-3.1.5/src/bin/exrcheck/main.cpp
@@ -13,6 +13,7 @@
 # include <unistd.h>
 #endif
 #include <vector>
+#include <cstdint>
 
 using namespace OPENEXR_IMF_NAMESPACE;
 using std::cout;
```

**Comment by meshula** [comment]
exrcheck/main.cpp already has that include

https://github.com/AcademySoftwareFoundation/openexr/blob/main/src/bin/exrcheck/main.cpp
```
#include <cstdint>
#include <fstream>
#include <iostream>
#include <string.h>
#if defined _WIN32 || defined _WIN64
#    include <io.h>
#else
#    include <unistd.h>
#endif
#include <vector>
```
Is it possible that there is a header ordering issue (I haven't seen one for cstdint before), or that there is another local fix you've got that fixed the problem? Or that you are building from a branch or a tag, missing the include?

**Comment by hobbes1069** [comment]
The problem was exposed on 3.1.5, was it fixed after that?

**Comment by meshula** [comment]
Ah, great question :)

https://github.com/AcademySoftwareFoundation/openexr/commit/c5763cfdf59ef4880dacdb11664782822a0976cd

It was fixed a couple of months after the release.

@cary-ilm This would seem to warrant a minor point release, or possibly a patch into 3.1.5. What do you think?

**Comment by hobbes1069** [comment]
Up to you guys. I have it fixed for Fedora until the next release. Not sure how many other distros are bleeding/leading edge when it comes to GCC. Any other significant fixes since the 3.1.5 release?

**Comment by cary-ilm** [comment]
Thanks for the catch. We've accumulated several fixes, we'll get a point release out shortly.

**Comment by cary-ilm** [comment]
This should be fixed in 3.1.6, thanks.
